### PR TITLE
[@angular/localize] add new $localize definitions (#46160)

### PR DESCRIPTION
**Pull request by sorohan**
This is a pretty weird one, and I'm not actually sure how best to handle it. Any advice is appreciated.

Angular *does* provide types for this package but because they are putting something on the global (`$localize`) it seems to be what `compilerOptions.types` is meant for, right? From typescript's docs:

> automatic inclusion is only important if you’re using files with global declarations (as opposed to files declared as modules)

But it doesn't make sense to re-declare all the other types from angular's package.

Please fill in this template.

- [ x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x] Test the change in your own code. (Compile and run.)
- [ x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## angular__localize ([<kbd>unpkg</kbd>](https://unpkg.com/browse/angular__localize@latest/))
was missing the following properties:
1. ɵcomputeMsgId
1. clearTranslations
1. loadTranslations
1. ɵMissingTranslationError
1. ɵfindEndOfBlock

<details>
<summary>as well as these 8 other properties...</summary>
<p>ɵisMissingTranslationError, ɵmakeParsedTranslation, ɵmakeTemplateObject, ɵparseMessage, ɵparseMetadata, ɵparseTranslation, ɵsplitBlock, ɵtranslate</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 89815e1859ae96ba77442c3792e69caecc39029a
</p>

**Comment by peterblazejewicz** [comment]
@sorohan
Please explain the reason why this is expected to exists in plain english.
There is a package with same soruce code, same namespace, what is that addition supposed to improve (except of problems with lazy loaded modules, which has no relation to DT, which is compile time thing):
https://github.com/angular/angular/blob/master/packages/localize/src/localize/src/localize.ts
I believe accepted way to introduce some namespace that belongs to other project, is to have someone from that original project to sign-off the change, to state clearly why a new source with widely exepcted namespace is being used in other place. There was some similar namespace-related content shipped recently
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/42392

Some of DT packages ships with their own README.md, to provide insight (and clear statement) on how to use the DT package and why it was introduced in the first place, see e.g.:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-bootstrap/README.md

Also note, that because of how TSC resolves dependency, there could be a lot of confiusion. See the bot outcome.
Sorry for posting concerns, my personal opinion.|

Maybe a note added to original angular.io, with a snippet on how to add agumented module definition, would be just fine and enough? Sample:
https://www.typescriptlang.org/docs/handbook/declaration-merging.html#global-augmentation
https://dev.to/chadalen/typescript-module-augmentation-in-angular-3o9n
